faunadb-scala/src/test/scala/faunadb/ClientSpec.scala

```diff
@@ -277,6 +276,114 @@ class ClientSpec
+      case e: Throwable => fail("Unexpected exception encountered!", e)
+    }
+  }
+
+  it should "accept tags if customer provides them" in {
+      val valueResponse: Value = client.query(Now(),
+                                              new RequestParameters(
+                                                tags = scala.collection.immutable
+                                                            .Map[String, String]("Key1" -> "Value1",
+                                                                                 "Key2" -> "Value2"))
+                                              )
+                                       .futureValue
+    valueResponse.toString should include("Z")
+  }
+
+  it should "throw error if provided tags contain invalid characters, 0x20" in {
+    try {
+      client.query(Now(),
+                   new RequestParameters(
+                     tags = scala.collection.immutable.Map[String, String]("Invalid Key" -> "Value1"))
+                   )
+            .futureValue
+      fail("Expected exception here, but none encountered")
+    } catch {
+      case re: RuntimeException => re.getMessage should include("contains invalid characters")
+      case e: Throwable => fail("Unexpected exception encountered!", e)
+    }
+  }
+
+  it should "throw error if provided tags contain invalid characters, special characters" in {
+    try {
+      client.query(Now(),
+                   new RequestParameters(
+                     tags = scala.collection.immutable.Map[String, String]("Tag@!---" -> "Value1"))
+                   )
+            .futureValue
+      fail("Expected exception here, but none encountered")
+    } catch {
+      case re: RuntimeException => re.getMessage should include("contains invalid characters")
+      case e: Throwable => fail("Unexpected exception encountered!", e)
+    }
+  }
+
+  it should "throw error if provided tags have a key with greater than 40 characters" in {
+    try {
+      client.query(Now(),
+                   new RequestParameters(
+                     tags = scala.collection
+                                 .immutable
+                                 .Map[String, String](Random.alphanumeric.take(41).mkString -> "Value1"))
+                   )
+            .futureValue
+      fail("Expected exception here, but none encountered")
+    } catch {
+      case re: RuntimeException => re.getMessage should include("longer than the allowable limit")
+      case e: Throwable => fail("Unexpected exception encountered!", e)
+    }
+  }
+
+  it should "throw error if provided tags have a value with greater than 80 characters" in {
+    try {
+      client.query(Now(),
+                   new RequestParameters(
+                     tags = scala.collection
+                                 .immutable
+                                 .Map[String, String]("Key1" -> Random.alphanumeric.take(81).mkString))
+                   )
```

Comment:
can we ahve a test for invalid chars in the value too?

Reply:
Added